lib/utmp.c

```diff
@@ -162,33 +162,34 @@ get_current_utmp(pid_t main_pid)
 			continue;
 
 		if (main_pid == ut->ut_pid) {
-			if (is_my_tty(ut->ut_line))
+			bool has_tty_match;
+
+			has_tty_match = is_my_tty(ut->ut_line);
+
+			if (NULL == ut_copy) {
+				ut_copy = XMALLOC(1, struct utmpx);
+				*ut_copy = *ut;
+			} else if (   has_tty_match
+			           || ut_copy_is_by_line_only)
+				*ut_copy = *ut;
+
+			if (has_tty_match)
```

Comment:
Please separate refactors from the actual fix into two separate commits.

Reply:
It could be hard as deep copy needs more allocations so it would be refactors anyway.
However, I may try to fully reproduce the old behaviour as the first commit.

Reply:
I wasn't really hard. Updated.

Reply:
Thanks a lot!